src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -289,6 +289,60 @@ internal ValueTask UnsubscribeAsync(int sid)
         return default;
     }
 
+    private static NatsOpts ReadUserInfoFromConnectionString(NatsOpts opts)
+    {
+        var first = true;
+
+        var natsUris = opts.GetSeedUris();
+        var maskedUris = new List<string>(natsUris.Length);
+
+        foreach (var natsUri in natsUris)
+        {
+            var uriBuilder = new UriBuilder(natsUri.Uri);
+
+            if (uriBuilder.UserName is { Length: > 0 } username)
+            {
+                if (first)
```

Comment:
we should alter all the uris same as the first one in case passwords on multiple uris are entered.